server/py/framework/utils/time_window_tracker.py

```diff
@@ -141,3 +130,12 @@ async def run_with_time_window_tracker(
             cycle_tracker.update_window,
             now,
         )
+        ensure_window_update = False
```

Comment:
why was it removed from creating the window update?

Reply:
the window update just above (lines 128-132). once done, there is no need to update again in the finally close. It's a insurance that window update will happen if the callback failed but it's not always wanted `_generate_events`

Reply:
Maybe add this as a comment so the reasoning here is clear?